shotgun_api3/shotgun.py

```diff
@@ -1470,7 +1470,8 @@ def upload(self, entity_type, entity_id, path, field_name=None,
         if not os.path.isfile(path):
             raise ShotgunError("Path must be a valid file, got '%s'" % path)
 
-        is_thumbnail = (field_name == "thumb_image" or field_name == "filmstrip_thumb_image")
+        is_thumbnail = (field_name == "thumb_image" or field_name == "filmstrip_thumb_image"
```

Comment:
Manne's referring to something along the lines of : 
is_thumbnail = (field_name in ["image", "thumb_image", "filmstrip_thumb_image", "filmstrip_image"])